### ISSUE TITLE
Redirection is improper with versioning

### ISSUE THREAD

**Issue by abhiTronix** [jimporter/mike#57]
Hello @jimporter,

I have discovered this annoying bug with versioning that make permalinks useless. Long story short, Mike [drop the hash fragment or any `#heading` (this is the versioning example)](https://github.com/squidfunk/mkdocs-material-example-versioning/blob/fbe5be4c1bc64c2c2bb6719b551059e244ad2d47/latest/index.html#L10). 

### Expected behavior

Redirected address from `latest` to  some `v0.1.2` version drops the `#heading` fragment  from url.

### Actual behavior

Must redirect to exact address without dropping  `#heading`  fragment.

### Steps to reproduce the bug

https://squidfunk.github.io/mkdocs-material-example-versioning/latest/#project-layout is redirected to https://squidfunk.github.io/mkdocs-material-example-versioning/0.2/ instead of https://squidfunk.github.io/mkdocs-material-example-versioning/0.2/#project-layout 

### Possible fix:

This could be solved with this new logic:

```javascript
<script>
    window.location.href.replace('/latest/','/<your version>/');
</script>
```
But I'm not sure. Please look into this.

### Related Issue
https://github.com/squidfunk/mkdocs-material/issues/2647

**Comment by jimporter** [comment]
Good point! Fixed in fd3c089. Between now and when v1.0.1 is released, you can just disable redirects (while will make full copies of the site for each alias) via `--no-redirect`, or copy the fixed template from `mike/templates/redirect.html` to a local dir and pass it to mike with `--template`.

**Comment by abhiTronix** [comment]
@jimporter Wow, I didn't about this feature. Also Kudos for fixing this bug. 😃

**Comment by abhiTronix** [comment]
> Adding a 404 page is complicated, but I hope to do it for 2.0.

@jimporter Yeah. For temporary solution, I redirect every 404 traffic to latest `404` page.

> This should work for the root index.html too, provided you run mike set-default again (#52 tracks keeping index.html up-to-date if the redirect template ever changes).

Yes, it is working. I just tested it. Thanks 😃
